blackcellmagic.py

```diff
@@ -37,10 +37,21 @@ def black(self, line, cell):
         args = magic_arguments.parse_argstring(self.black, line)
         line_length = args.line_length
         if cell:
-            formated = format_str(src_contents=cell, line_length=line_length)
-            if formated and formated[-1] == "\n":
+            # Comment magics
+            cell = re.sub(r"^%", r"##!%", cell)
+            cell = re.sub(r"\n\s*%", "\n##!%", cell)
+            try:
```

Comment:
This is a good logic! Can we abstract it out to a method and add tests for that method. It would be amazing if you can do that. Also, we should avoid exception based flow control.